include/cuco/detail/static_map_kernels.cuh

```diff
@@ -90,13 +92,13 @@ __global__ void insert(
   __shared__ typename BlockReduce::TempStorage temp_storage;
   std::size_t thread_num_successes = 0;
 
-  auto tid = block_size * blockIdx.x + threadIdx.x;
-  auto it  = first + tid;
+  auto const n = thrust::distance(first, last);
+  auto idx     = block_size * blockIdx.x + threadIdx.x;
```

Comment:
Let's be explicit about this type instead of implicitly using uint32_t
```suggestion
  int64_t idx     = block_size * blockIdx.x + threadIdx.x;
```

Reply:
Should this be `uint64_t`?

Reply:
I have the same question, since `n` is of `std::size_t` type, why is `int64_t` preferred over `uint64_t`/`std::size_t` here? By doing so, we do type conversions all the time for `idx < n`.

Reply:
`cuco::extent::value_type` will fix this problem soonish™️

Reply:
No, don't use `uint64_t`. Unsigned types are bad for loop variable performance.